libraries/pulse/src/consumer.js

```diff
@@ -3,6 +3,23 @@ const amqplib = require('amqplib');
 const assert = require('assert');
 const slugid = require('slugid');
 
+/**
+ * Recognize some "expected", ignorable errors due to normal network failures.
+ */
+const isExpectedError = err => {
```

Comment:
Everything about this function makes me uneasy but it is not your fault at all.

Reply:
I hate that there isn't a better way to do any of this.

Reply:
`amqplib` certainly leaves a great deal to be desired in terms of error handling.